From fe6a22b68c40902e5bdf52c3c29979ec0eb58747 Mon Sep 17 00:00:00 2001
From: Andreas Schwab <schwab@redhat.com>
Date: Wed, 15 Jun 2011 21:46:20 -0400
Subject: [PATCH] Improve 64-bit build

An imperfect version of fxstatat64 was used in ld.so.
(cherry picked from commit ee22793dd0b6c6bcbf341b5324da1af83d07dd13)
---
 ChangeLog                                          |    4 ++++
 .../unix/sysv/linux/wordsize-64/dl-fxstatat64.c    |    6 ++++++
 2 files changed, 10 insertions(+), 0 deletions(-)
 create mode 100644 sysdeps/unix/sysv/linux/wordsize-64/dl-fxstatat64.c

diff --git a/sysdeps/unix/sysv/linux/wordsize-64/dl-fxstatat64.c b/sysdeps/unix/sysv/linux/wordsize-64/dl-fxstatat64.c
new file mode 100644
index 0000000..339e5f4
--- /dev/null
+++ b/sysdeps/unix/sysv/linux/wordsize-64/dl-fxstatat64.c
@@ -0,0 +1,6 @@
+/* In this implementation we do not really care whether the call fails
+   because of missing kernel support since we do not even call the
+   function in this case.  */
+#undef __ASSUME_ATFCTS
+#define __ASSUME_ATFCTS 1
+#include "fxstatat.c"
-- 
1.7.6

